supervisor/jobs/__init__.py

```diff
@@ -1,49 +1,56 @@
 """Supervisor job manager."""
+from collections.abc import Callable
+from contextlib import contextmanager
+from contextvars import ContextVar, Token
 import logging
+from uuid import UUID, uuid4
+
+from attrs import define, field
+from attrs.setters import frozen
+from attrs.validators import ge, le
 
 from ..coresys import CoreSys, CoreSysAttributes
+from ..exceptions import JobNotFound, JobStartException
 from ..utils.common import FileConfiguration
 from .const import ATTR_IGNORE_CONDITIONS, FILE_CONFIG_JOBS, JobCondition
 from .validate import SCHEMA_JOBS_CONFIG
 
-_LOGGER: logging.Logger = logging.getLogger(__package__)
-
+current_job: ContextVar[UUID] = ContextVar("current_job")
 
-class SupervisorJob(CoreSysAttributes):
-    """Supervisor running job class."""
+_LOGGER: logging.Logger = logging.getLogger(__name__)
 
-    def __init__(self, coresys: CoreSys, name: str):
-        """Initialize the JobManager class."""
-        self.coresys: CoreSys = coresys
-        self.name: str = name
-        self._progress: int = 0
-        self._stage: str | None = None
 
-    @property
-    def progress(self) -> int:
-        """Return the current progress."""
-        return self._progress
+@define
+class SupervisorJob:
+    """Supervisor running job class."""
 
```

Comment:
Can you add a bit of documentation what this class exactly does?

E.g. Tracks a single Supervisor Job. Ensures that only one job is started on a single asyncio task globally, but allows child jobs even on the same asyncio task.

Ideally we should document a bit the big picture how the Job system exactly works, but I don't think that should be here in code.

Reply:
I added some more info to the `start` method as that made more sense to me. I also put some comments explaining why the contextvar works. Let me know if that works.